pymc/gp/cov.py

```diff
@@ -343,7 +353,7 @@ class Kron(Covariance):
     implementations.
     """
 
-    def __init__(self, factor_list):
+    def __init__(self, factor_list) -> None:
```

Comment:
Nitpick, but python never allows `__init__` to return anything, so it must always be `None`

Reply:
My reasoning was as per [this](https://stackoverflow.com/questions/64933298/why-should-we-use-in-def-init-self-n-none) i.e. it allows mypy to ensure `__init__` didn't return anything.

Reply:
According to this comment in that thread you linked, mypy assumes None by default: https://stackoverflow.com/a/67693892

Reply:
Seems it assumes `None` *provided* one or more of the args are type-annotated. The PEP advises 

> Note that the return type of __init__ ought to be annotated with -> None. [...]

However, I don't feel too strongly about it, so happy to remove these annotations.